requirements.txt

```diff
@@ -35,6 +33,6 @@ flake8-tabs == 2.1.0
 nose==1.3.7

```

Comment:
The version of Nose we're using is incompatible with Python 3.11, and it seems no new version is available. [Nose3](https://pypi.org/project/nose3/) is an alternative, though I'm not sure to what extend it can be used as a drop-in replacement.

Reply:
Nose3 doesn't support python 3.11. Nose2 is an alternative but I think dropping nose for native unittesting and moving to pytest if required is ideal.

Reply:
Nose2 doesn't explicitly support 3.11 either

Reply:
I've dropped nose as a dependency. It seems like we no longer need its test discovery